packages/sanity/src/_internal/cli/actions/schema/deleteSchemaAction.ts

```diff
@@ -0,0 +1,106 @@
+import {type CliCommandArguments, type CliCommandContext} from '@sanity/cli'
+import chalk from 'chalk'
+
+import {type ManifestWorkspaceFile} from '../../../manifest/manifestTypes'
+import {
+  getManifestPath,
+  readManifest,
+  SCHEMA_STORE_ENABLED,
+  throwIfProjectIdMismatch,
+} from './storeSchemasAction'
+
+export interface DeleteSchemaFlags {
+  'ids': string
+  'manifest-dir': string
+  'dataset': string
+}
+
+export default async function deleteSchemaAction(
+  args: CliCommandArguments<DeleteSchemaFlags>,
+  context: CliCommandContext,
+): Promise<void> {
+  if (!SCHEMA_STORE_ENABLED) {
+    return
+  }
+  const flags = args.extOptions
+  if (typeof flags.dataset === 'boolean') throw new Error('Dataset is empty')
+  if (typeof flags['manifest-dir'] === 'boolean') throw new Error('Manifest directory is empty')
+  if (typeof flags.ids !== 'string') throw new Error('--ids is required')
```

Comment:
Why do we assert booleans in some cases and not string in others? At least pick one style (id prefer !== string)

You could add a tiny assertString(parameterName, errorMessage) helper function and use for all of them imo.